### DISCUSSION TITLE
Duda recorrección Parte 2

### DISCUSSION THREAD

**Discussion by MaxiPis** [IIC2133-PUC/2024-2#194] [Tarea 3] [answered]
Hola qué tal?, escribo porque tengo una duda sobre la recorrección y me gustaría saber si me pueden ayudar. El tema es que no sé si entendí bien el enunciado o simplemente entendí mal y debo corregir mi código por el form. Va explicación…

En el enunciado del problema 2 parte explicando a grandes rasgos el input y dice “_Hay N maestros numerados del 1 al N, cada uno con conocimientos especiales en una técnica_”, yo entiendo que **siempre** se encontrarán N maestros del 1 a N y que, luego, serán las posibles conexiones entre maestros las que debemos considerar para determinar si es posible formar o no el MST. 

Luego, al revisar los testcases de la entrega, me di cuenta que todos me corrieron bien, salvo el easy-1 oohh nuuu. Analizando bien cual fue el error me di cuenta que el **input no contiene a todos los penguinos** y, por tanto, el output esperado es -1, sin embargo y según veo esto contradice el mismo enunciado que dice que hay N maestros numerados del 1 al N. El input es el siguiente y falta el penguin 2:
<div align="center">
<img src="https://github.com/user-attachments/assets/de693db6-d6df-4b8c-8c13-3b0b846eeb9e" alt="Evidencia" width="600" height="300">
</div>

Por otro lado, mi programa lo que hace es partir del supuesto que siempre el input tendrá a los N maestros numerados del 1 al N, de modo que no compruebo que esten los N maestros correlativos sino que simplemente veo si de todas las sesiones entre maestros se puede formar el MST. En el caso anterior mi respuesta sería 11:
<div align="center">
<img src="https://github.com/user-attachments/assets/28578615-6561-4e30-ab6c-8936149358f3" alt="Segunda evidencia" width="400" height="300">
</div>

Finalmente me gustaría agregar que al revisar el excel de las notar parece que otros también entendieron lo mismo que yo porque somos 5 que tenemos hasta hard-3 correctamente y solo nos falló easy-1. Entonces ¿leí mal o fue un error del test privado? 

No se si logré explicarme bien, igual quedo atento a cualquier aclaración.
<div align="center">
<img src="https://github.com/user-attachments/assets/00195ec7-708e-4ff8-90d0-7abd076325a7" alt="s evidencia" width="600" height="150">
</div>

**Comment by Segismundios** [comment, marked as answer]
Hola @MaxiPis. Tal y como dices, siempre habrá N maestros enumerados del 1 hasta el N. Esto quiere decir que siempre existen estos "nodos", por lo que se debería asumir siempre su existencia y, por lo tanto, es necesario corroborar si es posible conectarlo en el MST. Sin embargo, esto no quiere decir que siempre estén expresamente en el input, ya que ahí encontrarás las aristas que conectan un maestro con otro, no los maestros en sí. En este caso, el maestro 2 existe, sin embargo no tiene conexión con ningún otro maestro, por eso no está explícitamente en el input.
Espero que con esto se aclare tu duda, quedo atento, saludos!

**Comment by MaxiPis** [comment]
Ahhh, ahora entiendo la diferencia. Gracias por la aclaración :)
